### DISCUSSION TITLE
Question: Is there any way to get a system's belt to tighten up?

### DISCUSSION THREAD

**Discussion by Zitchas** [endless-sky/endless-sky#10069] [Balance, Mechanics, and Interactions]
Just trying to create an actual real asteroid belt, as in something that forms a fairly solid ring around the solar system, and the asteroids just appear almost everywhere. Sure, they're denser towards the center of the belt distance, and very little inside the habitable zone, but they just appear many multiples of the belt distance outside of it.

**Comment by Azure3141** [comment]
#8574 would have done this, but we decided to wait on a more thorough overhaul to asteroids.

**Reply by Zitchas** [reply]
Thanks for pointing that out. That PR would have done exactly what I am looking for.

**Comment by Zitchas** [comment]
As an example, here's what I was hoping we had a setting to achieve.

This was done using the very blunt hammer of finding L171-173. in mineables.cpp, which in vanilla looks like this:
```
	double sMin = max(.4 * (1. + eccentricity), .8 * (1. - eccentricity));
	double sMax = min(4. * (1. - eccentricity), 1.3 * (1. + eccentricity));
	orbitScale = (sMin + Random::Real() * (sMax - sMin)) * beltRadius;
```

and replacing it with this:
`orbitScale = beltRadius;`

I feel kind of bad about get rid of all those fancy formulas which probably took a lot of work to work out, but if it means I can actually make asteroid *belts* that provide useful tactical terrain, then it's worth it.

![Asteroid Orbits in an actual belt](https://github.com/endless-sky/endless-sky/assets/32169904/f44b828c-2413-42e6-9a2e-69ddba94747b)

That being said, I like the looks of Azure3141's PR. It'd be great to have the flexibility to have a tight belt without forcing *all* belts to be narrow rings.

**Reply by Zitchas** [reply]
I don't usually use the display things on all the NPC ships, but anyway, just thought I'd share a fullscreen image instead of just a piece of it.

Here's what arriving in Sol looks like in Delta with my current PR. That's 1020 mineables to make up a belt that dense at a radius of 4189.
![Asteroid Orbits in an actual belt 2](https://github.com/endless-sky/endless-sky/assets/32169904/eaaac04f-757f-4bcd-995f-b9d91d54aded)
